test/extended/openstack/egressip.go

```diff
@@ -227,6 +217,103 @@ spec:
+		}
+
+		e2e.Logf("Getting the worker for the EgressIP")
+		worker := workerNodeList.Items[0]
+
+		g.By(fmt.Sprintf("Getting the EgressIP network from the '%s' annotation", egressIPConfigAnnotationKey))
+		egressIPNetCidrStr, err := getEgressIPNetwork(worker, "ipv6")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(egressIPNetCidrStr).NotTo(o.BeEmpty(), "Could not get the EgressIP network from the '%s' annotation", egressIPConfigAnnotationKey)
+		e2e.Logf("Found the EgressIP network: %s", egressIPNetCidrStr)
+		ipv6mode, err := getipv6ModeFromSubnetCidr(networkClient, egressIPNetCidrStr)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if ipv6mode != "dhcpv6-stateful" {
+			e2eskipper.Skipf("Test not applicable for '%s' ipv6mode (only valid for '%s')", ipv6mode, "dhcpv6-stateful")
+		}
+
+		// Label the worker node as EgressIP assignable node
+		g.By(fmt.Sprintf("Labeling the primary node '%s' with '%s'", worker.Name, egressAssignableLabelKey))
+		node.AddOrUpdateLabelOnNode(clientSet, worker.Name, egressAssignableLabelKey, "dummy")
+		defer node.RemoveLabelOffNode(clientSet, worker.Name, egressAssignableLabelKey)
+
+		g.By("Obtaining a not in use IP address from the EgressIP network (machineNetwork cidr)")
+		machineNetworkID, err := getNetworkIdFromSubnetCidr(networkClient, egressIPNetCidrStr)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(machineNetworkID).NotTo(o.BeEmpty(), "Could not get the EgressIP network ID in openstack for '%s' subnet CIDR", egressIPNetCidrStr)
+		e2e.Logf("Found the EgressIP network ID '%s' in Openstack for the EgressIP CIDR '%s'", machineNetworkID, egressIPNetCidrStr)
+
+		egressIPAddrStr, err := getNotInUseEgressIP(networkClient, egressIPNetCidrStr, machineNetworkID)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(egressIPAddrStr).NotTo(o.BeEmpty(), "Couldn't find a free IP address in '%s' network in Openstack", egressIPNetCidrStr)
+		o.Expect(isIpv6(egressIPAddrStr)).To(o.BeTrue(), "egressIP should be IPv6 but it's %q", egressIPAddrStr)
+		egressIPAddrExtendedStr := FullIPv6(net.ParseIP(egressIPAddrStr))
+		e2e.Logf("Found '%s' free IP address in the EgressIP network in Openstack", egressIPAddrStr)
+
+		g.By("Creating a temp directory")
+		egressIPTempDir, err := os.MkdirTemp("", "egressip-e2e")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		e2e.Logf("Created '%s' temporary directory", egressIPTempDir)
+		defer os.RemoveAll(egressIPTempDir)
+
+		g.By("Creating an EgressIP yaml file")
+		var egressIPname = "egress-ip"
+		var egressIPYamlFileName = "egressip.yaml"
+		var egressIPYamlFilePath = egressIPTempDir + "/" + egressIPYamlFileName
+		var egressIPYamlTemplate = `apiVersion: k8s.ovn.org/v1
+kind: EgressIP
+metadata:
+  name: %s
+spec:
+  egressIPs:
+  - %s
+  namespaceSelector:
+    matchLabels:
+      %s`
+
+		egressIPYaml := fmt.Sprintf(egressIPYamlTemplate, egressIPname, egressIPAddrStr, "app: egress")
+		e2e.Logf("egressIPYaml: %s", egressIPYaml)
```

Comment:
This looks like something that could be extracted into a common function for all these tests, maybe even more of it.

Reply:
I agree

Reply:
done